### ISSUE TITLE
no_implicit_prelude attribute is not validated properly with unused_attributes lint

### ISSUE THREAD

**Issue by ehuss** [rust-lang/rust#81104]
The `no_implicit_prelude` attribute can only be applied to modules and the crate root. However, it seems to be silently ignored in all other positions:

```rust
#[no_implicit_prelude]
fn f() {
    #[no_implicit_prelude]
    {
        #[no_implicit_prelude]
        let s = String::new(); // OK
    }
}
```

I expected to see this happen: The UNUSED_ATTRIBUTES lint should trigger for all positions where it is ignored (with a future-incompatible note, since this should never have been allowed), possibly transitioned to a hard error in the future.

Instead, this happened: Attribute is ignored without a warning.

NOTE: There is [a test](https://github.com/rust-lang/rust/blob/8a6518427e11e6dd13d6f39663b82eb4f810ca05/src/test/ui/feature-gates/issue-43106-gating-of-builtin-attrs.rs#L399-L416) which checks this, but it is only firing because there is already a crate-level `no_implicit_prelude` attribute.

`rustc 1.51.0-nightly (c97f11af7 2021-01-10)`
